packages/xstate-wallet/src/channel-wallet.ts

```diff
@@ -286,12 +303,39 @@ export class ChannelWallet {
     }
   }
 
-  approveRichObjective(channelId: string) {
+  public async approveRichObjective(channelId: string): Promise<void> {
     const richObjective = this.store.richObjectives[channelId];
     if (!richObjective) {
       throw new Error(`Rich objective must exist for channelId ${channelId}`);
     }
-    this.crankRichObjectives({type: 'Approval'});
+
+    await this.crankRichObjectives({type: 'Approval'});
+
+    // TODO: need a better way to figure out when to broadcast an objective
+    //        since we don't know here if we are the creator of the objective or received the objective from elsewhere
+    if (!richObjective.myIndex) {
+      // TODO: generalize to other objectives
+      const objectiveToSend: OpenChannel = {
+        type: 'OpenChannel',
+        participants: richObjective.openingState.participants,
+        data: {targetChannelId: channelId, fundingStrategy: 'Direct'}
+      };
+
+      // TODO: we might want to first crank the objective, gather all states that need to be sent, and then send
+      // both the objective and the new states
```

Comment:
What's the reason not to do that in this PR?

Reply:
It is not clear to me how to batch together response data (akin to the `response` object in the server-wallet). I suspect this work will involve a somewhat significant refactor.